src/intelli/node/finddirectchild.cpp

```diff
@@ -77,48 +80,28 @@ FindDirectChildNode::eval()
     auto parent = nodeData<ObjectData>(m_in);
     if (!parent)
     {
-        gtTrace() << "FindDirectChildNode" << tr("Invalid parent");
         setNodeData(m_out, nullptr);
-        return;
+        return evalFailed();
     }
 
     auto const children = parent->object()->findDirectChildren();
 
     auto iter = std::find_if(std::begin(children), std::end(children),
-                             [this](GtObject const* c)
-    {
-        bool classCheck =
-        m_childClassName.get() == c->metaObject()->className();
-
-        bool nameCheck = true;
-        if (!m_objectName.getVal().isEmpty())
-        {
-            nameCheck = m_objectName.get() == c->objectName();
-        }
-
-        return classCheck && nameCheck;
+                             [this](GtObject const* c) {
+        auto const& targetClass = m_targetClassName.get();
+        bool classMatch = targetClass == c->metaObject()->className();
```

Comment:
But the logic is not the same as you describe the desired functionality. Here if the class name is not same no object is found. And as far as I can tell the class name is only updated if the class name edit in the widget is changed. (I have refactored the function but not the logic)

Reply:
Lets discuss this via phone on your next working day

Reply:
@mariusalexander I adapted the code to fix the unclear behavior. Now it should do what I wanted it to do. Feel free to make a cleanup if you see ugly lines ;)

Reply:
Looks good! 👍